### DISCUSSION TITLE
Adding encryption with git-remote-gcrypt

### DISCUSSION THREAD

**Discussion by gatoniel** [CharlesChiuGit/Logseq-Git-Sync-101#20] [Ideas]
As the contents of logseq graphs might be highly private, I am syncing my graphs with git-remote-gcrypt on a private repo on GitHub. I wonder if you're interested in also adding encryption in your 101. It should be straightforward as git-remote-gcrypt only encrypts when pushing and decrypts when pulling. However, encryption comes with exchanging gpg keys between machines and needs some understanding of gpg.

If you think this might be a good idea, I'd be happy to help.

**Comment by charliie-dev** [comment]
Interesting! I never used `git-remote-gcrypt` before. Is [this](https://github.com/spwhitton/git-remote-gcrypt) the correct repo?

Also, is `git-remote-gcrypt` available on all platform(windows, macos, ios, android and linux)? 

Although this 101 repo is suppose to be newbie-friendly, i guess it's also fine to include an advanced section in wiki?
PRs are always welcome! However, github can't directly do a PR on wiki page. Here's some workarounds for u to reference.
https://stackoverflow.com/questions/10642928/how-can-i-make-a-pull-request-for-a-wiki-page-on-github
https://www.growingwiththeweb.com/2016/07/enabling-pull-requests-on-github-wikis.html

**Comment by charliie-dev** [comment]
We might need to include:
1. a brief intro of gpg key.
2. how to setup a gpg key(or a github blog link).
3. steps/commands to use gcrypt.
4. things that u should be award of.
5. basic error handling about gcrypt.

**Comment by gatoniel** [comment]
Hi,
yes, I think this is the repo. I've heard issues about discontinued development, etc. But for me it works. I also cannot assess its security, but it is also referenced by git-annex.

I think it is only available on linux, which might be problematic in the scope of your tutorials.

I have written down a tutorial that I will put into better shape and add here, once I have time to do that. However, as you suggested, this might indeed be more a topic for an advanced section.

**Comment by charliie-dev** [comment]
Nice! Thanks for your effort! Looking forward to it!
